src/Symfony/Component/Dsn/ConnectionFactory.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Dsn;
+
+use Symfony\Component\Dsn\Exception\InvalidArgumentException;
+use Symfony\Component\Dsn\Factory\MemcachedFactory;
+use Symfony\Component\Dsn\Factory\RedisFactory;
+
+/**
+ * Factory for undetermined Dsn.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+final class ConnectionFactory
+{
+    const TYPE_REDIS = 'redis';
+    const TYPE_MEMCACHED = 'memcached';
+
+    /**
+     * @param string $dsn
+     *
+     * @return string
+     */
+    public static function getType($dsn)
+    {
+        if (!is_string($dsn)) {
+            throw new InvalidArgumentException(sprintf('The %s() method expect argument #1 to be string, %s given.', __METHOD__, gettype($dsn)));
+        }
+        if (0 === strpos($dsn, 'redis://')) {
+            return static::TYPE_REDIS;
+        }
+        if (0 === strpos($dsn, 'memcached://')) {
+            return static::TYPE_MEMCACHED;
+        }
+
+        throw new InvalidArgumentException(sprintf('Unsupported Dsn: %s.', $dsn));
+    }
+
+    /**
+     * Create a connection for a redis Dsn.
```

Comment:
This PhpDoc looks wrong -> `Create a connection for a redis Dsn.`

Reply:
done